### ISSUE TITLE
nextflow 24 fails to submit slurm jobs, unclear what the issue is.

### ISSUE THREAD

**Issue by d3v-null** [nextflow-io/nextflow#4931]
### Expected behavior and actual behavior

I expect nextflow 24 to behave the same way as nextflow 23 in terms of submitting slurm jobs. If it fails to submit a slurm job for some reason, there should be a clear error message with some indication of what the issue is.

### Steps to reproduce the problem

on a slurm cluster, this will submit jobs (which will probably fail, but it still submits them)

```bash
echo 9999999999 > obsids.csv
export MWA_ASVO_API_KEY='' PAWSEY_PROJECT='' 
nextflow run https://github.com/MWATelescope/MWAEoR-Pipeline -r main -profile genericHPC
```

but when I do the same thing with `v24.03.0-edge`

```bash
wget https://github.com/nextflow-io/nextflow/releases/download/v24.03.0-edge/nextflow-24.03.0-edge-all
chmod +x ./nextflow-24.03.0-edge-all
./nextflow-24.03.0-edge-all run https://github.com/MWATelescope/MWAEoR-Pipeline -r main -profile genericHPC
```

i get ...

### Program output 

```txt
Apr.-19 16:39:09.728 [Actor Thread 26] INFO  nextflow.processor.TaskProcessor - [skipping] Stored process > ws:wsMeta (1)
Apr.-19 16:39:09.768 [Actor Thread 29] DEBUG nextflow.processor.TaskProcessor - Process `ws:tapMeta (1)` is unable to find [UnixPath]: `/fred/oz048/dev/deleteme/nfdata/meta/1088451216_tapmeta.json` (pattern: `1088451216_tapmeta.json`)
Apr.-19 16:39:09.796 [Actor Thread 29] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]
Apr.-19 16:39:09.805 [Actor Thread 29] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:null, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.8-lunar; base
Packages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:null, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m
30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null, preserveFileTimestamp:null)
Apr.-19 16:39:09.889 [Task submitter] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for
  task: name=ws:tapMeta (1088451216); work-dir=/fred/oz048/dev/deleteme/work/a2/a9df0e462124c87b9b5124edad9868
  error [nextflow.exception.ProcessFailedException]: Error submitting process 'ws:tapMeta (1088451216)' for execution
Apr.-19 16:39:09.896 [Task submitter] INFO  nextflow.processor.TaskProcessor - [a2/a9df0e] NOTE: Error submitting process 'ws:tapMeta (1088451216)' for execution -- Error is ignored
Apr.-19 16:44:13.034 [Task monitor] DEBUG n.processor.TaskPollingMonitor - !! executor slurm > No more task to compute -- The following nodes are still active:
```

the workdir is empty

### Environment 

* Nextflow version: v24.03.0
* Java version: Temurin-11.0.20+8
* Operating system: Rocky Linux 9.3 (Blue Onyx)
* Bash version: Bash 5.1.8(1)-release (x86_64-redhat-linux-gnu)

### Additional context

Same thing happens on 

Java/17, SLES 12-SP5, Bash 4.3.48

**Comment by d3v-null** [comment]
it just occurred to me that setting `errorStrategy = 'terminate'` would provide some more info. it's some kind of issue handling GStrings

```
ERROR ~ Error executing process > 'ws:tapMeta (9999999999)'

Caused by:
  class org.codehaus.groovy.runtime.GStringImpl cannot be cast to class java.lang.String (org.codehaus.groovy.runtime.GStringImpl is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')

Source block:
  #!/usr/bin/env python3
  from pyvo.dal import TAPService
  import json
  
  tap = TAPService("http://vo.mwatelescope.org/mwa_asvo/tap")
  table = tap.search("SELECT * FROM mwa.observation WHERE obs_id = ${obsid}").table
  row = table[0]
  
  with open("${tapmeta}", "w") as f:
      json.dump(dict(row), f, indent=4, default=str)

Work dir:
  /pawsey/mwa/mwaeor/dev/deleteme/work/4d/54b0b062112893b5e0d27207f48dfe

Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line

 -- Check '.nextflow.log' file for details
 ````

**Comment by pditommaso** [comment]
Can you please include the `.nextflow.log` file?

**Comment by d3v-null** [comment]
[nextflow.log](https://github.com/nextflow-io/nextflow/files/15036349/nextflow.log)
@pditommaso sure thing.

**Comment by pditommaso** [comment]
Unfortunately not much more details. Can you please running nextflow as show below, and including the logs again: 

```
nextflow -trace nextflow run <your run options>
```

**Comment by d3v-null** [comment]
[nextflow-trace.log](https://github.com/nextflow-io/nextflow/files/15036549/nextflow-trace.log)
@pditommaso sure

**Comment by pditommaso** [comment]
This is indeed more useful 

```
nextflow.exception.ProcessFailedException: Error submitting process 'ws:tapMeta (1088451216)' for execution
	at nextflow.executor.GridTaskHandler.submit(GridTaskHandler.groovy:282)
	at nextflow.processor.TaskPollingMonitor.submit(TaskPollingMonitor.groovy:196)
	at nextflow.processor.TaskPollingMonitor.submitPendingTasks(TaskPollingMonitor.groovy:565)
	at nextflow.processor.TaskPollingMonitor.submitLoop(TaskPollingMonitor.groovy:390)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:343)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:328)
	at groovy.lang.MetaClassImpl.doInvokeMethod(MetaClassImpl.java:1333)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1088)
	at groovy.lang.MetaClassImpl.invokeMethodClosure(MetaClassImpl.java:1017)
	at groovy.lang.MetaClassImpl.doInvokeMethod(MetaClassImpl.java:1207)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1088)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1007)
	at groovy.lang.Closure.call(Closure.java:433)
	at groovy.lang.Closure.call(Closure.java:412)
	at groovy.lang.Closure.run(Closure.java:505)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassCastException: class org.codehaus.groovy.runtime.GStringImpl cannot be cast to class java.lang.String (org.codehaus.groovy.runtime.GStringImpl is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')
	at nextflow.executor.SlurmExecutor.getDirectives(SlurmExecutor.groovy:95)
	at nextflow.executor.AbstractGridExecutor.getDirectives(AbstractGridExecutor.groovy:128)
	at nextflow.executor.AbstractGridExecutor.getHeaders(AbstractGridExecutor.groovy:100)
	at nextflow.executor.AbstractGridExecutor.getHeaderScript(AbstractGridExecutor.groovy:84)
	at nextflow.executor.AbstractGridExecutor.createBashWrapperBuilder(AbstractGridExecutor.groovy:79)
	at nextflow.executor.GridTaskHandler.createTaskWrapper(GridTaskHandler.groovy:213)
	at nextflow.executor.GridTaskHandler.submit(GridTaskHandler.groovy:258)
	... 19 common frames omitted
```

**Comment by pditommaso** [comment]
The error can be replicated using this config snippet 

```
process.clusterOptions = { "-l h_vmem=${task.memory.toString().replaceAll(/[\sB]/,'')}" }
process.executor = 'slurm'
```

**Comment by pditommaso** [comment]
Solved by 74004fbd
